### PR TITLE
cleanup(quickstart): gkeconnect quickstart use correct inputs

### PR DESCRIPTION
**Pull request by cuiy0006**
Make gkeconnect quickstart use correct inputs.

`name=projects/*/locations/*/memberships/*`

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14666)

### CODE REVIEW

File: google/cloud/gkeconnect/quickstart/quickstart.cc [hunk 1]
```diff
@@ -14,21 +14,27 @@
 
 //! [all]
 #include "google/cloud/gkeconnect/gateway/v1/gateway_control_client.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
```
Comment (dbolduc) [inline google/cloud/gkeconnect/quickstart/quickstart.cc:17]: We don't want customers including internal headers in their applications, so we shouldn't include internal headers in our examples/quickstarts.

File: google/cloud/gkeconnect/CMakeLists.txt [hunk 2]
```diff
@@ -26,8 +26,10 @@ if (BUILD_TESTING AND GOOGLE_CLOUD_CPP_ENABLE_CXX_EXCEPTIONS)
     google_cloud_cpp_add_common_options(gkeconnect_quickstart)
     add_test(
         NAME gkeconnect_quickstart
-        COMMAND cmake -P "${PROJECT_SOURCE_DIR}/cmake/quickstart-runner.cmake"
-                $<TARGET_FILE:gkeconnect_quickstart> GOOGLE_CLOUD_PROJECT)
+        COMMAND
+            cmake -P "${PROJECT_SOURCE_DIR}/cmake/quickstart-runner.cmake"
+            $<TARGET_FILE:gkeconnect_quickstart> GOOGLE_CLOUD_PROJECT
+            GOOGLE_CLOUD_CPP_TEST_REGION "fakedmembership")
```
Comment (dbolduc) [inline google/cloud/gkeconnect/CMakeLists.txt:32]: The CMake helper function is actually accepting variables as input. So this thing will look up the variable called "fakedmembership".

Feel free to use any other variable, maybe with a comment saying that we don't expect it to be valid anyway.

File: google/cloud/gkeconnect/quickstart/quickstart.cc [hunk 3]
```diff
@@ -14,21 +14,27 @@
...
+
   namespace gkeconnect = ::google::cloud::gkeconnect_gateway_v1;
   auto client = gkeconnect::GatewayControlClient(
       gkeconnect::MakeGatewayControlConnection());
 
   google::cloud::gkeconnect::gateway::v1::GenerateCredentialsRequest request;
-  request.set_name(argv[1]);
+  std::string const name =
+      absl::StrCat(location.FullName(), "/memberships/", argv[3]);
+  request.set_name(name);
```
Comment (dbolduc) [inline google/cloud/gkeconnect/quickstart/quickstart.cc:37]: I'd just build the string inside `set_name(...)`

File: google/cloud/gkeconnect/quickstart/quickstart.cc (hunk 1 above)
Comment (cuiy0006) [reply google/cloud/gkeconnect/quickstart/quickstart.cc:17]: It is interesting, I am using `#include "absl/strings/str_cat.h"`, it is automatically replace by `checkers-pr`.

File: google/cloud/gkeconnect/CMakeLists.txt (hunk 2 above)
Comment (cuiy0006) [reply google/cloud/gkeconnect/CMakeLists.txt]: done

File: google/cloud/gkeconnect/quickstart/quickstart.cc (hunk 3 above)
Comment (cuiy0006) [reply google/cloud/gkeconnect/quickstart/quickstart.cc]: done.

File: google/cloud/gkeconnect/quickstart/quickstart.cc (hunk 1 above)
Comment (cuiy0006) [reply google/cloud/gkeconnect/quickstart/quickstart.cc]: https://screenshot.googleplex.com/AVbJ46XqY6yxaPG
Does that mean we can't use `absl::StrCat` in quickstart code?
I can replace it with plain string addition.

Comment (dbolduc) [reply google/cloud/gkeconnect/quickstart/quickstart.cc]: Yeah. I think we tend to do string addition in these.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**